YarnSpinner/VirtualMachine.cs

```diff
@@ -328,10 +336,22 @@ public enum ExecutionState {
                 /// - Stop
                 /** Immediately stop execution, and report that fact.
                  */
-                nodeCompleteHandler (new Dialogue.NodeCompleteResult (null));
-                executionState = ExecutionState.Stopped;
 
+                if (_returnStates.Count > 0)
```

Comment:
I'm conscious of the fact that this is redefining the meaning of "stop" but I'm not sure where or how else to have injected this unwind. Happy for suggestions from on high,